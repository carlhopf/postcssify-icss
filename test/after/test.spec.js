const browserify = require('browserify');
const rawBody = require('raw-body');
const postcssify2 = require('../../index');
const path = require('path');

describe('after', function () {
	var jsStream;
	var cssStream;
	
	beforeEach(() => {
		var bundle = browserify();
		
		bundle.plugin(postcssify2, {
			after: [
				require('postcss-calc'),
			],
		});
		
		bundle.add(path.join(__dirname, 'index.js'));
		bundle.on('css stream', (stream) => { cssStream = stream; });
		jsStream = bundle.bundle();
	});
	
	beforeEach((cb) => {
		rawBody(jsStream, (err, buf) => {
			expect(err).toBeFalsy();
			cb(null);
		});
	});
	
	test('postcss-calc', (cb) => {
		rawBody(cssStream, (err, buf) => {
			expect(err).toBeFalsy();
			expect(buf.toString('utf8')).toMatchSnapshot();
			cb(null);
		});
	});
});

describe('after', function () {
	var jsStream;
	var cssStream;
	
	beforeEach(() => {
		var bundle = browserify();
		
		bundle.plugin(postcssify2, {
			after: [],
		});
		
		bundle.add(path.join(__dirname, 'index.js'));
		bundle.on('css stream', (stream) => { cssStream = stream; });
		jsStream = bundle.bundle();
	});
	
	beforeEach((cb) => {
		rawBody(jsStream, (err, buf) => {
			expect(err).toBeFalsy();
			cb(null);
		});
	});
	
	test('no postcss-calc', (cb) => {
		rawBody(cssStream, (err, buf) => {
			expect(err).toBeFalsy();
			expect(buf.toString('utf8')).toMatchSnapshot();
			cb(null);
		});
	});
});
