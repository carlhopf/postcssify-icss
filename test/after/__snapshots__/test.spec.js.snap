// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`after no postcss-calc 1`] = `
"
.button {
  background-color: calc(2px + 2px);
}
"
`;

exports[`after postcss-calc 1`] = `
"
.button {
  background-color: 4px;
}
"
`;
